lib/local_subnet_manager.cpp

```diff
@@ -0,0 +1,12 @@
+#include "local_subnet_manager.hpp"
+#include "spa_message.hpp"
+#include "logical_address.hpp"
+
+void LocalSubnetManager::receiveMessage(SpaMessage message) 
```

Comment:
What if we made functions like this take a shared pointer to avoid so much copying?